MeshLib/Elements/Element.h

```diff
@@ -173,6 +173,14 @@ class Element
     virtual Element* clone() const = 0;
 
     /**
+     * Constructs a new object polymorphically. This is similar to clone, but
+     * accepts new nodes and id.
+     * \pre The length of the \c nodes vector is equal to the derived element's
+     * total number of nodes.
+     */
+    virtual Element* construct(Node** nodes, std::size_t id) const = 0;
```

Comment:
it doesn't construct itself. `clone()` also calls the constructor of the same type, isn't it? since the argument is different, you can still use the same function name.